client/components/webpack-build-monitor/index.tsx

```diff
@@ -0,0 +1,153 @@
+/**
+ * @see https://webpack.js.org/api/module-variables/#__webpack_hash__-webpack-specific
+ */
+declare const __webpack_hash__: string;
+
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import Spinner from 'components/spinner';
+import { IntervalHandle, TimestampMS } from 'client/types';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const CONNECTION_TIMEOUT = 20 * 1000;
+const debug = debugFactory( 'calypso:webpack-build-monitor' );
+
+enum BuildState {
+	UNKNOWN,
+	IDLE,
+	BUILDING,
+	BUILT_WITH_WARNINGS,
+	ERROR,
+}
+
+const WebpackBuildMonitor: FunctionComponent = () => {
+	const [ lastHash, setLastHash ] = useState( __webpack_hash__ );
+	const [ isConnected, setIsConnected ] = useState( false );
+	const [ buildState, setBuildState ] = useState( BuildState.UNKNOWN );
+
+	useEffect( () => {
+		if ( typeof EventSource === 'undefined' ) {
+			if ( process.env.NODE_ENV !== 'production' ) {
+				// eslint-disable-next-line no-console
+				console.warn( 'Webpack build monitor disabled. No `EventSource`.' );
+			}
+		} else {
```

Comment:
There can be an early `return` here. Saves one level of indentation for the rest of the (big) function 🙂